packages/webapp/src/containers/Animals/Inventory/useAnimalInventory.tsx

```diff
@@ -0,0 +1,343 @@
+  return t(`TYPE.${key}`, { ns: 'animal' });
+};
+
+const getAnimalBreedLabel = (key: string) => {
+  return t(`BREED.${key}`, { ns: 'animal' });
+};
+
+const chooseAnimalIdentifier = (animal: Animal) => {
+  if (animal.name && animal.identifier) {
+    return `${animal.name} | ${animal.identifier}`;
+  } else if (animal.name && !animal.identifier) {
+    return animal.name;
+  } else if (!animal.name && animal.identifier) {
+    return animal.identifier;
+  } else {
+    return `${t('ANIMAL.ANIMAL')}#${animal.internal_identifier}`;
+  }
+};
+
+const chooseAnimalTypeLabel = (
+  animalOrBatch: Animal | AnimalBatch,
+  defaultAnimalTypes: DefaultAnimalType[],
+  customAnimalTypes: CustomAnimalType[],
+) => {
+  if (animalOrBatch.default_type_id) {
+    return getAnimalTypeLabel(
+      getProperty(defaultAnimalTypes, animalOrBatch.default_type_id, 'key'),
+    );
+  } else if (animalOrBatch.custom_type_id) {
+    return getProperty(customAnimalTypes, animalOrBatch.custom_type_id, 'type');
+  } else {
+    return null;
+  }
+};
+
+const chooseAnimalBreedLabel = (
+  animalOrBatch: Animal | AnimalBatch,
+  defaultAnimalBreeds: DefaultAnimalBreed[],
+  customAnimalBreeds: CustomAnimalBreed[],
+) => {
+  if (animalOrBatch.default_breed_id) {
+    return getAnimalBreedLabel(
+      getProperty(defaultAnimalBreeds, animalOrBatch.default_breed_id, 'key'),
+    );
+  } else if (animalOrBatch.custom_breed_id) {
+    return getProperty(customAnimalBreeds, animalOrBatch.custom_breed_id, 'type');
+  } else {
+    return null;
+  }
+};
+
+const sortAnimalsData = (
+  animals: Animal[],
+  animalGroups: AnimalGroup[],
+  customAnimalBreeds: CustomAnimalBreed[],
+  customAnimalTypes: CustomAnimalType[],
+  defaultAnimalBreeds: DefaultAnimalBreed[],
+  defaultAnimalTypes: DefaultAnimalType[],
+) => {
+  return animals.map((animal: Animal) => {
+    return {
+      icon: animal.default_type_id
+        ? getDefaultAnimalIcon(defaultAnimalTypes, animal.default_type_id)
+        : CattleIcon,
+      identification: chooseAnimalIdentifier(animal),
+      type: chooseAnimalTypeLabel(animal, defaultAnimalTypes, customAnimalTypes),
+      breed: chooseAnimalBreedLabel(animal, defaultAnimalBreeds, customAnimalBreeds),
+      groups: animal.group_ids
+        ? animal.group_ids.map((id: number) => getProperty(animalGroups, id, 'name'))
+        : ['none'],
+      path: `/animal/${animal.internal_identifier}`,
+      count: 1,
+    };
+  });
+};
+
+const sortAnimalBatchesData = (
+  animalBatches: AnimalBatch[],
+  animalGroups: AnimalGroup[],
+  customAnimalBreeds: CustomAnimalBreed[],
+  customAnimalTypes: CustomAnimalType[],
+  defaultAnimalBreeds: DefaultAnimalBreed[],
+  defaultAnimalTypes: DefaultAnimalType[],
+) => {
+  return animalBatches.map((batch: AnimalBatch) => {
+    return {
+      icon: BatchIcon,
+      identification: batch.name
+        ? batch.name
+        : `${t('ANIMAL.ANIMAL')}#${batch.internal_identifier}`,
```

Comment:
Okay I can do that -- since it was simpler I was leaving it but yeah I can use the same function. 

An | Or might have worked here too to be simple.